cli/checkpoint.go

```diff
@@ -0,0 +1,92 @@
+package main
+
+import (
+	"context"
+	"encoding/hex"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"math/big"
+	"strconv"
+
+	eigenpodproofs "github.com/Layr-Labs/eigenpod-proofs-generation"
+	"github.com/Layr-Labs/eigenpod-proofs-generation/cli/onchain"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/fatih/color"
+)
+
+func RunCheckpointProof(ctx context.Context, eigenpodAddress string, eth *ethclient.Client, chainId *big.Int, beaconClient BeaconClient, out, owner *string) {
+	lastCheckpoint := lastCheckpointedForEigenpod(eigenpodAddress, eth)
```

Comment:
`currentCheckpoint` is probably a better name for this, since we have an EigenPod concept of `lastCheckpointTimestamp`, but that's not what we're dealing with here.